allensdk/brain_observatory/behavior/vbo_2023_nwb_writers/behavior/schemas.py

```diff
@@ -0,0 +1,176 @@
+        ),
+    )
+    driver_line = List(
+        String,
+        required=True,
+        cli_as_single_argument=True,
+        description="Genetic driver line(s) of subject",
+    )
+    equipment_name = String(
+        required=True, description=("Name of the equipment used.")
+    )
+    full_genotype = String(
+        required=True, description="Full genotype of subject"
+    )
+    mouse_id = String(
+        required=True,
+        description="LabTracks ID of the subject. aka external_specimen_name.",
+    )
+    project_code = String(
+        rquired=True,
+        description="LabTracks ID of the subject. aka external_specimen_name.",
+    )
+    reporter_line = String(
+        required=True, description="Genetic reporter line(s) of subject"
+    )
+    session_type = String(
+        required=True, description="Full name of session type."
+    )
+    sex = String(required=True, description="Subject sex")
+
+    @mm.post_load
+    def convert_date_time(self, data, **kwargs):
+        """Change date_of_acquisition to a date time type from string."""
+        data["date_of_acquisition"] = pd.to_datetime(
+            data["date_of_acquisition"], utc=True
+        )
+        return data
+
+
+class BaseInputSchema(ArgSchema):
+    class Meta:
+        unknown = mm.RAISE
+
+    log_level = LogLevel(
+        default="INFO", description="Logging level of the module"
+    )
+    metadata_table = InputFile(
+        required=True,
+        description="CSV file containing rows of BehaviorSession or "
+        " BehaviorOphysExperiment metadata. Must at least contain the "
+        "metadata of the session requested in relevant.",
+    )
+    skip_metadata_key = List(
+        String,
+        required=False,
+        cli_as_single_argument=True,
+        description="List of metadata table keys to skip. Can be used to "
+        "override known data issues. Example: ['mouse_id']",
+        default=[],
+    )
+    skip_stimulus_file_key = List(
+        String,
+        required=False,
+        cli_as_single_argument=True,
+        description="List of stimulus file keys to skip. Can be used to "
+        "override known data issues. Example: ['mouse_id']",
+        default=[],
+    )
+    output_path = String(
+        required=True,
+        validate=check_write_access_overwrite,
+        description="Path of output.json to be written",
+    )
+
+
+class BehaviorInputSchema(BaseInputSchema):
+
+    behavior_session_id = Int(
+        required=True, description="Id of BehaviorSession to create."
+    )
+
+    behavior_session_metadata = Nested(
+        BehaviorSessionMetadataSchema,
+        required=True,
+        description="Data pertaining to a behavior session",
+    )
+
+    @mm.pre_load
+    def retreive_metadata(self, data, **kwargs):
+        """Load the data from csv using Pandas the same as the
+        project_cloud api.
+        """
+        df = literal_col_eval(
```

Comment:
this code needs to be abstracted out. It is currently duplicated, as you say in project_cloud api